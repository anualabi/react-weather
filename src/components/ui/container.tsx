import { cn } from "@/lib/utils";

export type ContainerProps = Readonly<{
  children: React.ReactNode;
  className?: string;
}>;

export default function Container({ children, className }: ContainerProps) {
  return (
    <div className={cn("max-w-[1440px] w-full mx-auto px-5", className)}>
      {children}
    </div>
  );
}
