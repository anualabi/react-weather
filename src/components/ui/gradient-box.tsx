import { cn } from "@/lib/utils";

export type GradientBoxProps = Readonly<{
  children: React.ReactNode;
  className?: string;
}>;

export default function GradientBox({ children, className }: GradientBoxProps) {
  return (
    <div
      className={cn(
        "bg-gradient-to-b from-transparent from-0% via-[#C4C4C405] via-43.85% to-[#C4C4C420] to-100% rounded-md",
        className
      )}
    >
      {children}
    </div>
  );
}
