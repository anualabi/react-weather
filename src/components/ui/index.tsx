export { default as Container } from "./container";
export { H2 } from "./typography";
