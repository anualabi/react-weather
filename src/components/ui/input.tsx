import { forwardRef, InputHTMLAttributes } from "react";

import { cn } from "@/lib/utils";

export type InputProps = InputHTMLAttributes<HTMLInputElement>;

const Input = forwardRef<HTMLInputElement, InputProps>(
  ({ className, type, ...props }, ref) => {
    return (
      <input
        type={type}
        className={cn(
          "flex h-12 w-full rounded-md border border-black bg-secondary px-3 py-1 text-white text-sm lg:text-base shadow-sm transition-colors placeholder:text-zinc-500 focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-zinc-950 disabled:cursor-not-allowed disabled:opacity-50",
          className
        )}
        ref={ref}
        {...props}
      />
    );
  }
);
Input.displayName = "Input";

export default Input;
