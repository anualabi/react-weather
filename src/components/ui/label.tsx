import { forwardRef, LabelHTMLAttributes } from "react";
import { cva, type VariantProps } from "class-variance-authority";

import { cn } from "@/lib/utils";

const labelVariants = cva(
  "text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70"
);

export type LabelProps = LabelHTMLAttributes<HTMLLabelElement> &
  VariantProps<typeof labelVariants> & {
    className?: string;
  };

const Label = forwardRef<HTMLLabelElement, LabelProps>(
  ({ children, className, ...props }, ref) => {
    return (
      <label ref={ref} className={cn(labelVariants(), className)} {...props}>
        {children}
      </label>
    );
  }
);

Label.displayName = "Label";

export default Label;
