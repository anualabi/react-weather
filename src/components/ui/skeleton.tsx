import { cn } from "@/lib/utils";

type SkeletonProps = Readonly<React.HTMLAttributes<HTMLDivElement>>;

export default function Skeleton({ className, ...props }: SkeletonProps) {
  return (
    <div
      data-testid="skeleton"
      className={cn("animate-pulse rounded-md bg-zinc-900/10", className)}
      {...props}
    />
  );
}
