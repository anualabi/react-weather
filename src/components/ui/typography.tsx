import { forwardRef, HTMLAttributes } from "react";

import { cn } from "@/lib/utils";

const H2 = forwardRef<HTMLHeadingElement, HTMLAttributes<HTMLHeadingElement>>(
  ({ children, className, ...props }, ref) => {
    return (
      <h2
        {...props}
        ref={ref}
        className={cn(
          "scroll-m-20 text-2xl md:text-3xl xl:text-4xl",
          className
        )}
      >
        {children}
      </h2>
    );
  }
);
H2.displayName = "H2";

export { H2 };
