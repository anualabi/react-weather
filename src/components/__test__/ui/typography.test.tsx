import { render, screen } from "@testing-library/react";

import { H2 } from "@/components/ui/typography";

const mockProps = {
  children: "Test Typography",
  className: "",
};

describe("H2", () => {
  it("should render (snapshot)", () => {
    const { asFragment } = render(<H2 {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render an h2 HTML element", () => {
    render(<H2 {...mockProps} />);
    const h2Element = screen.getByText("Test Typography");
    expect(h2Element.tagName).toBe("H2");
  });

  it("should render its children", () => {
    render(<H2 {...mockProps} />);
    const h2Element = screen.getByText(mockProps.children);
    expect(h2Element).toBeInTheDocument();
  });
});
