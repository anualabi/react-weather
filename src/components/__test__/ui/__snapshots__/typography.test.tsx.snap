// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H2 should render (snapshot) 1`] = `
<DocumentFragment>
  <h2
    class="scroll-m-20 text-2xl md:text-3xl xl:text-4xl"
  >
    Test Typography
  </h2>
</DocumentFragment>
`;
